### PR TITLE
Implement afterBlockExecute hook for the reward module - Closes #6690

### LINKED ISSUE #6690: Implement afterBlockExecute hook of Rewards Module
### Description
Implement `afterBlockExecute` hook as specified in https://research.lisk.com/t/define-state-transitions-of-reward-module/316#after-block-execution-22

### Acceptance criteria
- Possible scenarios are tested in the unit test(e.g. BFT violation, seedReveal violation etc.)

### Additional Information
- `TOKEN_ID_REWARD` will be specified in config parameter of init method of the reward module class.

### Related Issue
#6686

### PR DESCRIPTION
**Pull request by mehmetegemen**
### What was the problem?

This PR resolves #6690 

### How was it solved?

- Implemented afterBlockExecute hook in the reward module class

### How was it tested?

- It was tested with unit tests

### CODE REVIEW

File: framework/src/modules/reward/module.ts
Comment (shuse2) [inline framework/src/modules/reward/module.ts:72]: `0` reward should be valid, and it is calculating by itself, so no validation is needed?

File: framework/test/unit/modules/reward/reward_module.spec.ts
Comment (shuse2) [inline framework/test/unit/modules/reward/reward_module.spec.ts:59]: i don't think it's necessary to test all the brackets assuming the util is tested well

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: framework/src/modules/reward/module.ts
Comment (mehmetegemen) [reply framework/src/modules/reward/module.ts:72]: I thought minting 0 reward is useless and redundant, don't you think so? Because if I don't end the control flow there the API will mint 0 reward.

File: framework/test/unit/modules/reward/reward_module.spec.ts
Comment (mehmetegemen) [reply framework/test/unit/modules/reward/reward_module.spec.ts:59]: Then you think I should test the just first bracket in a single test?

File: framework/src/modules/reward/module.ts
Comment (mehmetegemen) [reply framework/src/modules/reward/module.ts:72]: On the other hand, 0 reward is valid and not minting is `mint` functions responsibility, so probably I should remove this conditional. Still waiting for your approval.

File: framework/test/unit/modules/reward/reward_module.spec.ts
Comment (shuse2) [reply framework/test/unit/modules/reward/reward_module.spec.ts:59]: yes, i think just first bracket after offset should be good

File: framework/src/modules/reward/module.ts
Comment (shuse2) [reply framework/src/modules/reward/module.ts:72]: yes, then it should be if `blockReward <= 0` return. if thrown error, block will be rejected

Comment (mehmetegemen) [reply framework/src/modules/reward/module.ts:72]: done

File: framework/test/unit/modules/reward/reward_module.spec.ts
Comment (mehmetegemen) [reply framework/test/unit/modules/reward/reward_module.spec.ts:59]: done

**Review by shuse2 — APPROVED** [review]

**Review by Incede — APPROVED** [review]

**Pull request merged**
